### PR TITLE
Total groups before sort

### PR DESCRIPTION
**Pull request by ataft**
Sort may rely on totals, so we need to call addTotals(groups) before groups.sort()

### CODE REVIEW

File: slick.dataview.js [hunk 1]
```diff
@@ -997,7 +1001,6 @@
       if (groupingInfos.length) {
         groups = extractGroups(newRows);
         if (groups.length) {
-          addTotals(groups);
```
Comment (ghiscoding) [inline slick.dataview.js:1000]: Simple, and maybe stupid question, if you keep that line there, does it count in double? 

The downside I can see is that it's on the `recalc`, if user ask for a recalculation, the total calculation should still be part of it, I assume... or do we? I see that the `refresh()` calls the `recalc()`, so I'm not sure what impact this would have... I'm not sure when that is used, maybe when user adds a new row? or update a row?

Comment (ataft) [reply slick.dataview.js:1000]: I don't think it will ever duplicate because addTotals is not cumulative, i.e. it assigns a new total every time, as seen here:
https://github.com/6pac/SlickGrid/blob/master/slick.dataview.js#L694

However, there's no purpose in leaving this line since I simply moved it into `extractGroups()`. This recalc() function is the only place that calls `extractGroups()`, so there's no concern that it might cause problems elsewhere.

Simply put, there's no code between `groups.sort()` and `addTotals(groups)`. All I did was put addTotals() first, which I can't imagine could possibly break anything.

Comment (ghiscoding) [reply slick.dataview.js:1000]: ok that was the only thing that I was curious about, I'm ok with the rest. I assume you tried with a few of the grouping examples to make sure nothing is broken? 

Let's wait to see if @6pac has anything to say or if wants to merge as is. I'm good with it.

Comment (ataft) [reply slick.dataview.js:1000]: Yes, but the grouping examples only use `a.count` or `a.title` in the `comparer(a,b)`. They don't use the totals, which is probably why this bug was never caught.
Thanks!
